salt/_grains/etcd_info.py

```diff
@@ -0,0 +1,40 @@
+try:
+    import etcd
+    HAS_LIBS = True
+except ImportError:
+    HAS_LIBS = False
+
+from salt.exceptions import CommandExecutionError
+
+def etcd_info():
+
+  etcd_info = {}
+  etcd_info['members_all'] = []
+  etcd_info['member_type'] = "proxy"
+
+  if HAS_LIBS:
+    etcd_info['available'] = True
+    client = etcd.Client(host='127.0.0.1', port=2379)
+  else:
+    etcd_info['available'] = False
+    raise CommandExecutionError(
+        '(unable to import etcd, '
+        'module most likely not installed)'
+    )
+
+  machine_file = open('/etc/machine-id', 'r')
+  machine_id = machine_file.read().rstrip()
+  machine_file.close()
```

Comment:
I think this block could be improved with:

```
with open("/etc/machine-id", "r") as machine_file:
    machine_id = machine_file.read().rstrip()
```
so the file would be closed automatically, even in case of error. And it would be even better if that could be surrounded by an exception catch...